flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/failover/ExecutionFailureHandlerTest.java

```diff
@@ -183,6 +194,55 @@ void testNonRecoverableFailureHandlingResult() throws Exception {
         assertThat(executionFailureHandler.getNumberOfRestarts()).isZero();
     }
 
+    /** Test isNewAttempt of {@link FailureHandlingResult} is expected. */
+    @Test
+    void testNewAttemptAndNumberOfRestarts() throws Exception {
```

Comment:
The comment is obsolete because it could be expressed through the test method name(s) (to reduce redundant information in the code). 

Additionally, what's your opinion on splitting this big test up into smaller test methods. I see the benefit of having one big one. Because you can check the number of restarts at the same time. But having more specific test methods usually helps with improving readability of the test code.. WDYT?

Reply:
Like in my previous comment: We can avoid using comments in favor of the assert message :shrug:

Reply:
Good suggestion!

Also, I extracted the `assertHandlerRootException` and `assertHandlerConcurrentException` methods, after that, the `testNewAttemptAndNumberOfRestarts` is totally simple.